src/webrtc/peerconnection.ts

```diff
@@ -629,6 +656,96 @@ export class PeerConnectionClass implements PeerConnection<signals.Message> {
   }
 }  // class PeerConnectionClass
 
+// Takes a DataChannel and provides a ControlChannel API atop of it.
+class ControlChannelWrapper implements datachannel.ControlChannel {
```

Comment:
This seems like a lot of machinery.

As I see it, `getControlChannel` is a hack to work around how hard it currently is, thanks to `Pool`, to create non-proxying datachannels. Fine, but I'd much rather see further effort go into removing `Pool` and, in the meantime, we can simply mark `getControlChannel` with:
 * `// WARNING: use this channel at your peril, it's used for heartbeat and lifecycle purposes`
 * `// TODO: remove this`

(the warning might be overkill: I can see how the `ControlChannel` interface helps, so long as it's a subset of `DataChannel`)